source/geometry_gmsh.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "geometry_common_includes.h"
+
+#include <deal.II/grid/grid_in.h>
+
+#include <fstream>
+
+namespace ryujin
+{
+  namespace Geometries
+  {
+    /**
+     * This class imports a triangulation from a GMSH *.msh file.
+     *
+     * @ingroup Mesh
+     */
+    template <int dim>
+    class GMSH : public Geometry<dim>
+    {
+    public:
+      GMSH(const std::string subsection)
+          : Geometry<dim>("gmsh", subsection)
+      {
+        filename_ = "ryujin.msh";
+        this->add_parameter("filename", filename_, "GMSH *.msh file to load");
```

Comment:
@gregorychristian I have changed your commit slightly by adding a parameter for the file that should be read in and renamed the class to `GMSH`.